duplicate key value JDBC exception when indexing large Mercurial repo

When indexing large internal Mercurial repo I hit this exception:

```
2013-11-04 11:48:39.341+0100 FINE t50 Executor.exec: Executing command [/usr/bin/hg, log, -f, --template, changeset: {rev}:{node|short}\n{branches}{tags}{parents}\nuser: {author}\ndate: {date|isod
ate}\nfiles: {files}\ndescription: {desc|strip|obfuscate}\n, usr/src/foo] in directory /var/opengrok/src/bar
2013-11-04 11:48:39.881+0100 WARNING t50 HistoryGuru.createCache: An error occured while creating cache for /var/opengrok/src/bar (MercurialRepository)
org.opensolaris.opengrok.history.HistoryException: java.sql.SQLIntegrityConstraintViolationException: The statement was aborted because it would have caused a duplicate key value in a unique or primary key constraint or unique index identified by 'FILECHANGES_PK' defined on 'FILECHANGES'.
        at org.opensolaris.opengrok.history.JDBCHistoryCache.store(JDBCHistoryCache.java:612)
        at org.opensolaris.opengrok.history.Repository.createCache(Repository.java:311)
        at org.opensolaris.opengrok.history.HistoryGuru.createCache(HistoryGuru.java:478)
        at org.opensolaris.opengrok.history.HistoryGuru.access$000(HistoryGuru.java:51)
        at org.opensolaris.opengrok.history.HistoryGuru$1.run(HistoryGuru.java:522)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:744)
Caused by: java.sql.SQLIntegrityConstraintViolationException: The statement was aborted because it would have caused a duplicate key value in a unique or primary key constraint or unique index identified by 'FILECHANGES_PK' defined on 'FILECHANGES'.
        at org.apache.derby.client.am.SQLExceptionFactory40.getSQLException(Unknown Source)
        at org.apache.derby.client.am.SqlException.getSQLException(Unknown Source)
        at org.apache.derby.client.am.PreparedStatement.executeUpdate(Unknown Source)
        at org.opensolaris.opengrok.history.JDBCHistoryCache.storeHistory(JDBCHistoryCache.java:826)
        at org.opensolaris.opengrok.history.JDBCHistoryCache.store(JDBCHistoryCache.java:607)
        ... 9 more
Caused by: org.apache.derby.client.am.SqlException: The statement was aborted because it would have caused a duplicate key value in a unique or primary key constraint or unique index identified by 'FILECHANGES_PK' defined on 'FILECHANGES'.
        at org.apache.derby.client.am.Statement.completeExecute(Unknown Source)
        at org.apache.derby.client.net.NetStatementReply.parseEXCSQLSTTreply(Unknown Source)
        at org.apache.derby.client.net.NetStatementReply.readExecute(Unknown Source)
        at org.apache.derby.client.net.StatementReply.readExecute(Unknown Source)
        at org.apache.derby.client.net.NetPreparedStatement.readExecute_(Unknown Source)
        at org.apache.derby.client.am.PreparedStatement.readExecute(Unknown Source)
        at org.apache.derby.client.am.PreparedStatement.flowExecute(Unknown Source)
        at org.apache.derby.client.am.PreparedStatement.executeUpdateX(Unknown Source)
        ... 12 more
2013-11-04 11:48:49.934+0100 INFO t1 Indexer.prepareIndexer: Done...
2013-11-04 11:48:49.936+0100 INFO t1 Indexer.doIndexerExecution: Starting indexing
```

This happened after the main `hg log` for the repo directory was done and `hg log` commands for renamed files were executed (in the above log snippet `usr/src/foo` is the file which was renamed in the past).

This is related to the recently added change for #22.

This is a regression and stopper.

--- a/src/org/opensolaris/opengrok/history/MercurialHistoryParser.java
+++ b/src/org/opensolaris/opengrok/history/MercurialHistoryParser.java
@@ -155,8 +155,9 @@ class MercurialHistoryParser implements Executor.StreamHandler {
                       */
                      String[] move = part.split(" \\(");
                      File f = new File(mydir + move[0]);
-                     if (!move[0].isEmpty() && f.exists()) {
-                        renamedFiles.add(move[0]);
+                     if (!move[0].isEmpty() && f.exists() &&
+                         !renamedFiles.contains(move[0])) {
+                             renamedFiles.add(move[0]);
                      }
                 }
 

